src/transactions/TransactionFrame.cpp

```diff
@@ -1682,6 +1781,317 @@ TransactionFrame::apply(AppConnector& app, AbstractLedgerTxn& ltx,
+
+        auto res = op->applyParallel(app, entryMap, config, sorobanConfig,
+                                     ledgerInfo, sorobanMetrics, opResult,
+                                     *sorobanData, txPrngSeed);
+
+#ifdef BUILD_TESTS
+        maybeTriggerTestInternalError(mEnvelope);
+#endif
+
+        if (res.getSuccess())
+        {
+            // Build OperationMeta
+            LedgerEntryChanges changes;
+            for (auto const& newUpdates : res.getModifiedEntryMap())
+            {
+                auto const& lk = newUpdates.first;
+                auto const& le = newUpdates.second;
+
+                // Any key the op updates should also be in entryMap because the
+                // keys were taken from the footprint (the ttl keys were added
+                // as well)
+                auto prev = entryMap.find(lk);
+                releaseAssertOrThrow(prev != entryMap.end());
+
+                auto prevLe = prev->second.mLedgerEntry;
+
+                if (prevLe)
+                {
+                    changes.emplace_back(LEDGER_ENTRY_STATE);
+                    changes.back().state() = *prevLe;
+
+                    if (le)
+                    {
+                        changes.emplace_back(LEDGER_ENTRY_UPDATED);
+                        changes.back().updated() = *le;
+                    }
+                    else
+                    {
+                        changes.emplace_back(LEDGER_ENTRY_REMOVED);
+                        changes.back().removed() = lk;
+                    }
+                }
+                else
+                {
+                    // op should return a LedgerEntry for this key if it doesn't
+                    // exist in EntryMap because it must've been created in the
+                    // op
+                    releaseAssertOrThrow(le);
+
+                    // New entry
+                    changes.emplace_back(LEDGER_ENTRY_CREATED);
+                    changes.back().created() = *le;
+                }
+
+                LedgerTxnDelta::EntryDelta entryDelta;
+                if (prevLe)
+                {
+                    entryDelta.previous =
+                        std::make_shared<InternalLedgerEntry>(*prevLe);
+                }
+                if (le)
+                {
+                    auto deltaLe = *le;
+                    // This is for the invariants check in LedgerManager
+                    deltaLe.lastModifiedLedgerSeq = ledgerInfo.getLedgerSeq();
+
+                    entryDelta.current =
+                        std::make_shared<InternalLedgerEntry>(deltaLe);
+                }
+
+                effects.getDelta().entry[lk] = entryDelta;
+                // Note that we don't set delta.header here because Soroban
+                // transactions don't modify the header. The header will be set
+                // right before we cal into the invariants.
+            }
+
+            if (op->getOperation().body.type() == RESTORE_FOOTPRINT)
```

Comment:
I think it might be easier just to build both the restore and op meta above and consolidate the logic. The only reason we had the post process function before was due to the limitations of ltx. Now that we have to build all meta ourselves anyway without the ltx, this post processing step is unnecessary. I think this may help simplify auto restore too.